test/tests/syncLocalTest.js

```diff
@@ -22,62 +22,108 @@ describe("Zotero.Sync.Data.Local", function() {
 			assert.isTrue(handled);
 			assert.isTrue(cont);
+			assert.isTrue(newAccountFile.exists());
+		});
+		
+		it("should prompt for data reset and alert if wrong confirmation text entered", function* (){
+			yield Zotero.Users.setCurrentUserID(1);
+			yield Zotero.Users.setCurrentUsername("A");
 			
-			assert.equal(Zotero.Users.getCurrentUserID(), 2);
-			assert.equal(Zotero.Users.getCurrentUsername(), "B");
-		})
+			var handled = false;
+			waitForDialog(function (dialog) {
+				dialog.document.getElementById('zotero-hardConfirmationDialog-textbox').value = 'B';
+				
+				waitForDialog(function (dialog) {
+					dialog.document.getElementById('zotero-hardConfirmationDialog-textbox').value = 'A';
+					
+					handled = true;
+				}, 'accept', 'chrome://zotero/content/hardConfirmationDialog.xul');
+				
+			}, 'accept', 'chrome://zotero/content/hardConfirmationDialog.xul');
+			waitForDialog();
+			var cont = yield Zotero.Sync.Data.Local.checkUser(window, 2, "B");
+			var newAccountFile = Zotero.getZoteroDirectory().clone();
+			newAccountFile.append('new-account');
+			assert.isTrue(handled);
+			assert.isTrue(cont);
+			assert.isTrue(newAccountFile.exists());
+		});
+
 		
-		it("should prompt for user update and cancel", function* () {
+		it("should prompt for data reset and cancel", function* () {
 			yield Zotero.Users.setCurrentUserID(1);
 			yield Zotero.Users.setCurrentUsername("A");
 			
-			waitForDialog(false, 'cancel');
-			var cont = yield Zotero.Sync.Data.Local.checkUser(null, 2, "B");
+			waitForDialog(false, 'cancel', 'chrome://zotero/content/hardConfirmationDialog.xul');
+			var cont = yield Zotero.Sync.Data.Local.checkUser(window, 2, "B");
+			var newAccountFile = Zotero.getZoteroDirectory().clone();
+			newAccountFile.append('new-account');
 			assert.isFalse(cont);
+			assert.isFalse(newAccountFile.exists());
 			
 			assert.equal(Zotero.Users.getCurrentUserID(), 1);
 			assert.equal(Zotero.Users.getCurrentUsername(), "A");
-		})
+		});
 		
-		it("should update local relations when syncing for the first time", function* () {
-			yield resetDB({
-				thisArg: this,
-				skipBundledFiles: true
-			});
-			
-			var item1 = yield createDataObject('item');
-			var item2 = yield createDataObject(
-				'item', { libraryID: Zotero.Libraries.publicationsLibraryID }
-			);
-			
-			yield item1.addLinkedItem(item2);
+		it("should prompt for data reset and allow to choose a new data directory", function* (){
+			sinon.stub(Zotero, 'forceNewDataDirectory').returns(true);
+			yield Zotero.Users.setCurrentUserID(1);
+			yield Zotero.Users.setCurrentUsername("A");
 			
-			var cont = yield Zotero.Sync.Data.Local.checkUser(null, 1, "A");
+			waitForDialog(null, 'extra1', 'chrome://zotero/content/hardConfirmationDialog.xul');
+			waitForDialog();
+			var cont = yield Zotero.Sync.Data.Local.checkUser(window, 2, "B");
+			var newAccountFile = Zotero.getZoteroDirectory().clone();
+			newAccountFile.append('new-account');
```

Comment:
Here and elsewhere, it's better to use OS.File for anything new instead of nsIFile, since we're more likely to mock OS.File in Electron and, regardless, something based on string paths will probably be easier to adapt. So, e.g., `OS.Path.join(Zotero.getZoteroDirectory().path, 'new-account');` (but not actually `new-account`, per the note above).